src/precice/impl/DataContext.cpp

```diff
@@ -104,6 +91,35 @@ bool DataContext::hasMapping() const
   return hasReadMapping() || hasWriteMapping();
 }
 
+bool DataContext::isMappingRequired()
+{
+  using namespace mapping;
+  if (not hasMapping()) {
+    return false;
+  }
+
+  auto timing    = _mappingContext.timing;
+  bool hasMapped = _mappingContext.hasMappedData;
+  bool mapNow    = timing == MappingConfiguration::ON_ADVANCE;
+  mapNow |= timing == MappingConfiguration::INITIAL;
```

Comment:
I understand that you just copied this implementation from the original code snippet(s), but I already stumbled across this once: For me the code would be much better readable using 
```suggestion
  const bool hasMapped = _mappingContext.hasMappedData;
  const bool mapNow    = (timing == MappingConfiguration::ON_ADVANCE) |= 
                         (timing == MappingConfiguration::INITIAL);
```
because it reveals the dependencies of `mapNow` more obvious. However, opinions might be different and I just mention it here for the sake of completeness.